src/runtime/task/waker.rs

```diff
@@ -28,6 +28,10 @@ unsafe fn raw_waker_clone(data: *const ()) -> RawWaker {
 unsafe fn raw_waker_wake(data: *const ()) {
     let task_id = TaskId::from(data as usize);
     ExecutionState::with(|state| {
+        if state.has_cleaned_up() {
```

Comment:
Yes, but specifically because dropping a channel from the futures crate can call a waker. In general though, a waker could be called by a drop implementation of any library afaik.

Reply:
If you comment out this if statement, the test `asynch::channel::oneshot_once` will fail. In the stack trace, you can see where `futures::channel::oneshot::drop_tx` calls the waker.